docs/customization/state_machine.rst

```diff
@@ -0,0 +1,122 @@
+Customizing State Machines
+==========================
+
+.. warning::
+    Not familiar with the State Machine concept? Read the docs :doc:`here </book/state_machine>`!
+
+How to customize a State Machine?
+---------------------------------
+
+.. tip::
+
+    First of all if you are attempting to change anything in any state machine in **Sylius** you will need a special file:
+    ``app/config/state_machine.yml`` which has to be imported in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    imports:
+        - { resource: "state_machine.yml" }
+
+How to add a new state?
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to add a new **state** to the state machine of :doc:`Sylius' Checkout </book/checkout>`.
+You will need to add these few lines to the ``state_machine.yml``:
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
+            states:
+                your_new_state: ~ # here name your state as you wish
+
+After that your new step will be available alongside other steps that already were definded in that state machine.
+
+How to add a new transition?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to add a new **transition** to the state machine of :doc:`Sylius' Checkout </book/checkout>`,
+that will allow moving from the ``completed`` state backwards to ``payment_selected``. Let's call it "cancelling".
+
+You will need to add these few lines to the ``state_machine.yml``:
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
+            transitions:
+                cancel:
+                    from: [completed]
+                    to: payment_selected
+
+After that your new transition will be available alongside other transitions that already were definded in that state machine.
+
+How to add a new callback?
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's assume that you would like to add a new **callback** to the state machine of :doc:`Sylius' Checkout </book/checkout>`,
+that will do something on an already defined transition.
+
+You will need to add these few lines to the ``state_machine.yml``:
+
+.. code-block:: yaml
+
+    # app/config/state_machine.yml
+    winzou_state_machine:
+        sylius_order_checkout:
```

Comment:
This example is perfect, but it should be ``sylius_order`` state machine, not checkout.